src/org/sosy_lab/java_smt/test/InterpolatingProverTest.java

```diff
@@ -305,34 +344,40 @@ public <T> void sequentialInterpolation() throws SolverException, InterruptedExc
 
     assertThat(stack).isUnsatisfiable();
 
-    List<BooleanFormula> itps1 = stack.getSeqInterpolants0(ImmutableList.of(TA, TB, TC, TD));
-    List<BooleanFormula> itps2 = stack.getSeqInterpolants0(ImmutableList.of(TD, TC, TB, TA));
-    List<BooleanFormula> itps3 = stack.getSeqInterpolants0(ImmutableList.of(TA, TC, TB, TD));
-    List<BooleanFormula> itps4 =
-        stack.getSeqInterpolants(
-            Lists.transform(ImmutableList.of(TA, TA, TA, TB, TC, TD, TD), ImmutableSet::of));
-    List<BooleanFormula> itps5 =
-        stack.getSeqInterpolants(
-            Lists.transform(ImmutableList.of(TA, TA, TB, TC, TD, TA, TD), ImmutableSet::of));
-    List<BooleanFormula> itps6 =
-        stack.getSeqInterpolants(
-            Lists.transform(ImmutableList.of(TB, TC, TD, TA, TA, TA, TD), ImmutableSet::of));
-
-    stack.close();
-
-    checkItpSequence(ImmutableList.of(A, B, C, D), itps1);
-    checkItpSequence(ImmutableList.of(D, C, B, A), itps2);
-    checkItpSequence(ImmutableList.of(A, C, B, D), itps3);
-    checkItpSequence(ImmutableList.of(A, A, A, B, C, D, D), itps4);
-    checkItpSequence(ImmutableList.of(A, A, B, C, D, A, D), itps5);
-    checkItpSequence(ImmutableList.of(B, C, D, A, A, A, D), itps6);
+    if (interpolationStrategy == null) {
+      List<BooleanFormula> itps1 = stack.getSeqInterpolants0(ImmutableList.of(TA, TB, TC, TD));
+      List<BooleanFormula> itps2 = stack.getSeqInterpolants0(ImmutableList.of(TD, TC, TB, TA));
+      List<BooleanFormula> itps3 = stack.getSeqInterpolants0(ImmutableList.of(TA, TC, TB, TD));
+      List<BooleanFormula> itps4 =
+          stack.getSeqInterpolants(
+              Lists.transform(ImmutableList.of(TA, TA, TA, TB, TC, TD, TD), ImmutableSet::of));
+      List<BooleanFormula> itps5 =
+          stack.getSeqInterpolants(
+              Lists.transform(ImmutableList.of(TA, TA, TB, TC, TD, TA, TD), ImmutableSet::of));
+      List<BooleanFormula> itps6 =
+          stack.getSeqInterpolants(
+              Lists.transform(ImmutableList.of(TB, TC, TD, TA, TA, TA, TD), ImmutableSet::of));
+
+      stack.close();
+
+      checkItpSequence(ImmutableList.of(A, B, C, D), itps1);
+      checkItpSequence(ImmutableList.of(D, C, B, A), itps2);
+      checkItpSequence(ImmutableList.of(A, C, B, D), itps3);
+      checkItpSequence(ImmutableList.of(A, A, A, B, C, D, D), itps4);
+      checkItpSequence(ImmutableList.of(A, A, B, C, D, A, D), itps5);
+      checkItpSequence(ImmutableList.of(B, C, D, A, A, A, D), itps6);
+    } else {
+      assertThrows(UnsupportedOperationException.class,
```

Comment:
This would imply that this is intented behavior. But we could implement sequential and/or tree interpolants. So i would simply put an assumption at the beginning of the method that does not run this test for solvers/options that don't support it. (see my comment about the `requireTreeInterpolation()` method) 
This goes for all occurrences of this in this class.